cmd/api/src/database/sso_providers.go

```diff
@@ -137,3 +138,10 @@ func (s *BloodhoundDB) GetSSOProviderUsers(ctx context.Context, id int) (model.U
 
 	return users, CheckError(s.db.WithContext(ctx).Table("users").Where("sso_provider_id = ?", id).Find(&users))
 }
+
+func (s *BloodhoundDB) GetSSOProviderById(ctx context.Context, id int) (model.SSOProvider, error) {
+	var provider model.SSOProvider
+	result := s.db.WithContext(ctx).Preload("OIDCProvider").Preload("SAMLProvider").Table("sso_providers").Where("id = ?", id).Find(&provider)
```

Comment:
You can use the table constant instead of `sso_providers`

You can also replace Find with .First since we expect only one SSO provider to exist with the ID.

I think we need an integration test as well